src/renderer/viz/expressions/stats.js

```diff
@@ -0,0 +1,29 @@
+
+/**
+ * Calculate Average
+ *
+ * @param {Number[]} values
+ * @returns {Number} - average
+ */
+export const average = (values) => values.reduce((sum, value) => sum + value, 0) / values.length;
+
+/**
+ * Calculate Variance
+ *
+ * @param {Number[]} values
+ * @param {Number} average
+ * @returns {Number[]} - calculate variance for each element in values
+ */
+export const variance = (values, average) => values.map(value => Math.pow(value - average, 2));
+
+/**
+ * Calculate Standard Deviation (STD), using population deviation formula
+ *
+ * @param {Number[]} values
+ * @returns {Number} - standard deviation
+ */
+export const standardDeviation = (values) => {
+    const avg = average(values);
```

Comment:
The viewport version will need to call this function per frame, and `values` can be a large array. This should probably be optimized, but first, it should be profiled.

In particular, I'm suspicious of the performance of `map`, `reduce` and `Math.pow`. Using an old-fashioned `for (let i=0 ...` and `x*x` instead of `Math.pow` will probably improve things.

Reply:
In the viewport version, I changed it to use its own private methods, to calculate average and std from histogram frequencies. But I agree that further optimizations could be applied. I'm gonna add another task in this PR. Thx!